Development/nmos/control_protocol_state.cpp

```diff
@@ -0,0 +1,361 @@
+#include "nmos/control_protocol_state.h"
+
+#include "nmos/control_protocol_methods.h"
+#include "nmos/control_protocol_resource.h"
+
+namespace nmos
+{
+    namespace experimental
+    {
+        namespace details
+        {
+            // create control class
+            //   where
+            //     properties: vector of NcPropertyDescriptor where NcPropertyDescriptor can be constructed using make_control_class_property
+            //     methods:    vector of NcMethodDescriptor vs assoicated method handler where NcMethodDescriptor can be constructed using make_nc_method_descriptor
+            //     events:     vector of NcEventDescriptor where NcEventDescriptor can be constructed using make_nc_event_descriptor
+            control_class make_control_class(const utility::string_t& description, const nc_class_id& class_id, const nc_name& name, const web::json::value& fixed_role, const std::vector<web::json::value>& properties_, const std::vector<nmos::experimental::method>& methods_, const std::vector<web::json::value>& events_)
```

Comment:
Related to comments in `node_implementation.cpp` I think we need to distinguish between the class definitions (descriptors) and the class instances.  It's not clear whether a function called `make_control_class` is creating a definition or an instance.

My suggested solution would be either:
1. maintain `make_control_class` here and append `_instance` to the functions that produce instances of classes e.g. `make_nc_worker_instance`
2. call this `make_control_class_descriptor` to indicate it's creating a definition (descriptor) and keep the instance names as they are
I would favour 2 I think.

Reply:
done as no. 2